src/classes/BDI_DataImportService.cls

```diff
@@ -612,9 +620,43 @@ global with sharing class BDI_DataImportService {
         }
     }
 
+    /**
+     * @description Before processing data import records, this method filters out any data imports that do not meet
+     * custom criteria that applies to all data imports before allowing them to be processed.
+     * @param dataImports {List<DataImport__c>} List of data import records to be filtered
+     *
+     * @return List<DataImport__c>
+    */
+    private List<DataImport__c> shouldProcess(List<DataImport__c> dataImports) {
+        List<DataImport__c> dataImportsToProcess = new List<DataImport__c>();
+
+        for (DataImport__c dataImport : dataImports) {
+            /* Use a single boolean to determine if each data import should be processed. If new criteria is added
+            * it can be added to isSuccessfulPayment using conditional operators. */
+            Boolean shouldProcess = isSuccessfulPayment(dataImport);
+
+            if (shouldProcess) {
+                dataImportsToProcess.add(dataImport);
+            }
+        }
+
+        return dataImportsToProcess;
+    }
+
+    private Boolean isSuccessfulPayment(DataImport__c dataImport) {
```

Comment:
Even though this method is private, I think a comment header to explain the behavior like we discussed would be nice

Reply:
updated!